pkg/prometheus/promcfg.go

```diff
@@ -2530,6 +2598,17 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 					Value: selectors,
 				})
 			}
+
+			if config.AttachMetadata != nil {
+				switch config.Role {
+				case "pod", "Pod":
+					configs[i] = cg.WithMinimumVersion("2.35.0").AppendMapItem(configs[i], "attach_metadata", config.AttachMetadata)
+				case "endpoints", "Endpoints", "endpointslice", "EndpointSlice":
+					configs[i] = cg.WithMinimumVersion("2.37.0").AppendMapItem(configs[i], "attach_metadata", config.AttachMetadata)
+				default:
+					level.Warn(cg.logger).Log("msg", fmt.Sprintf("ignoring attachMetadata not supported by Prometheus for role: %s", config.Role))
```

Comment:
should we reject the resource in the first place?

Reply:
I think its ok to warn than rejecting the resource. We do similar for other version specific newer fields.

Reply:
I'm on the fence too. Prometheus wouldn't fail either if you configure node's metadata with a service role so a warning is probably fine.
For clarity in the CRD description, it would be better to duplicate the struct from pkg/api/monitoring/v1 and state in the v1alpha1 comment that it's only valid for Pod, Endpoint and Endpointslice roles.